src/kem/kyber768.rs

```diff
@@ -97,28 +107,26 @@ pub fn decapsulate(
     let hashed = G(&to_hash);
     let (k_not, pseudorandomness) = hashed.split_at(32);
 
-    let expected_ciphertext_result =
+    // If a ciphertext C is well-formed, setting aside the fact that a
+    // decryption failure could (with negligible probability) occur, it must hold that:
+    //
+    // Encrypt(pk, Decrypt(sk, C)) = C
+    //
+    // Therefore, if |ind_cpa::encrypt| returns an error,
+    // |expected_ciphertext| cannot equal |ciphertext|, thereby resulting in
+    // implicit rejection.
+    //
+    // If C is ill-formed, due to the use of hashing to obtain |pseudorandomness|
+    // as well as the fact that the Kyber CPA-PKE is sparse pseudo-random, it is
+    // highly likely that |expected_ciphertext| will not equal |ciphertext|, thereby
+    // also resulting in implicit rejection.
+    //
+    // Thus, we ignore the second return value of |ind_cpa::encrypt|.
+    let (expected_ciphertext, _) =
```

Comment:
Should we put the implicit rejection value in the second part or something instead of the constant time select? No concrete idea though ...